google/cloud/forseti/services/inventory/base/cai_gcp_client.py

```diff
@@ -842,23 +842,23 @@ def iter_compute_vpntunnels(self, project_number):
     # Disabling pulling ke data from CAI since it's missing nodepools
     # attribute in the returned data. We can re-enable this once that
     # problem is resolved.
-    # def iter_container_clusters(self, project_number):
-    #     """Iterate Kubernetes Engine Cluster from Cloud Asset data.
-    #
-    #     Args:
-    #         project_number (str): number of the project to query.
-    #
-    #     Yields:
-    #         dict: Generator of Kubernetes Engine Cluster resources.
-    #     """
-    #     resources = self.dao.iter_cai_assets(
-    #         ContentTypes.resource,
-    #         'container.googleapis.com/Cluster',
-    #         '//cloudresourcemanager.googleapis.com/projects/{}'.format(
-    #             project_number),
-    #         self.session)
-    #     for cluster in resources:
-    #         yield cluster
+    def iter_container_clusters(self, project_number):
+        """Iterate Kubernetes Engine Cluster from Cloud Asset data.
```

Comment:
Does the cloud asset data include nodepools now in the cluster resource?

If not, you'll need to add container.googleapis.com/NodePool to the list of resources that CAI exports and then have a helper function that gets the node pools for a cluster and adds them to the nodePools key.

Something like (psuedocode)

```
for cluster in resources:
    node_pools = self._get_nodepools_for_cluster(cluster)
    cluster['nodePools'] = sorted(node_pools, key=lambda np: np['selfLink'])
    yield cluster
```

Reply:
CAI still doesn't include nodepools in the cluster resource. I commented out the cluster implementation in this file so it should fall back on the implementation in gcp.py. Let's re-enable this when CAI sends nodepools in the cluster resource.